src/index.js

```diff
@@ -19,9 +19,24 @@ import reducers from './reducers';
 import { UserContextProvider } from './context/UserContext';
 import PostContextProvider from './context/PostContext';
 
+// this is a persisted state from the previous usage of the app.
+const persistedState = localStorage.getItem('reduxState')
+  ? JSON.parse(localStorage.getItem('reduxState'))
+  : {};
+
 // This is the basic setup for Redux - we give create store the Combined reducers called 'reducer'
 // and compose is simply applying the middleware so we can perform async functions.
-const store = createStore(reducers, compose(applyMiddleware(thunk)));
+const store = createStore(
+  reducers,
+  persistedState,
+  compose(applyMiddleware(thunk)),
+);
+
+// subscribes to any changes in redux store and save it to local storage
+store.subscribe(() => {
+  // Save it in local storage so that if we refresh any page, we are still logged in.
+  localStorage.setItem('reduxState', JSON.stringify(store.getState()));
```

Comment:
I haven't seen it done like this. If you are confident and it works at your branch yeah let's do it.